inference-engine/src/mkldnn_plugin/mkldnn_edge.cpp

```diff
@@ -93,7 +93,7 @@ bool MKLDNNEdge::needReorder() {
     };
 
     const auto portChildEdges = getParent()->getChildEdgesAtPort(inNumber);
-    if (in_place && detectInPlaceChildsNum(portChildEdges) > 1 && childCanChangeMem)
+    if (in_place && childCanChangeMem && portChildEdges.size() > 1 && detectInPlaceChildsNum(portChildEdges) > 1)
```

Comment:
Could you please explain why we need this change? Are you sure it will not brake other scenarios?

Reply:
This is actually useful feature that helps to avoid insertion of unnecessary reorders. In the old configuration, there were situations when the algorithm would insert reorder on an edge that used inplace memory, even though there was only one such edge between the two layers. As far as I understand, this change should not be harmful. But on the other hand, it does not lead to a visible performance improvement, and if it seems to be dangerous, it can be removed.

Reply:
Discussed offline. Additional accuracy checks will be performed.